model/task_lifecycle.go

```diff
@@ -1245,6 +1241,10 @@ func MarkOneTaskReset(t *task.Task, logIDs bool) error {
 			if err != nil {
 				return errors.Wrap(err, "retrieving execution task")
 			}
+			// Ignore succeeded execution tasks if we only want to restart failed.
+			if utility.FromBoolPtr(t.RestartFailed) && !evergreen.IsFailedTaskStatus(execTask.Status) {
```

Comment:
Should also check ResetWhenFinished isn't set though; if the user did ask to restart the whole thing then it would make sense to do it all at once

Reply:
i think you were talking about the RestartFailed field? (which has been renamed to ResetFailedWhenFinished) so I started marking them groups